--- vpb-driver-4.2.58/src/utils/dtmfcheck.cpp.orig	2014-09-28 20:02:28.000000000 +0200
+++ vpb-driver-4.2.58/src/utils/dtmfcheck.cpp	2019-12-01 17:11:16.935351767 +0100
@@ -49,7 +49,7 @@
 	printf("  --block size  Granularity of analysis.  Default is 16000 samples (2sec)\n");
 } //}}}
 
-static double dBFactor( double db ) { return pow10( db / 20.0 ); }
+static double dBFactor( double db ) { return pow( 10, db / 20.0 ); }
 
 static inline void LinearGain( double gain, int16_t *buf, size_t len )
 { //{{{
