--- vpb-driver-4.2.56/src/vtcore/vtopenpci.c~	2013-04-11 19:11:39.000000000 +0200
+++ vpb-driver-4.2.56/src/vtcore/vtopenpci.c	2013-06-05 22:56:18.849529470 +0200
@@ -262,13 +262,13 @@
 	return (const char*)ent->driver_data;
 }
 
-static int __devinit openpci_probe_board(struct pci_dev *pdev, const struct pci_device_id *ent);
-static void __devexit openpci_remove_board(struct pci_dev *pdev);
+static int openpci_probe_board(struct pci_dev *pdev, const struct pci_device_id *ent);
+static void openpci_remove_board(struct pci_dev *pdev);
 
 static struct pci_driver vtopenpci_driver = {
 	.name     = NAME,
 	.probe    = openpci_probe_board,
-	.remove   = __devexit_p(openpci_remove_board),
+	.remove   = openpci_remove_board,
 	.suspend  = NULL,
 	.resume   = NULL,
 	.id_table = openpci_idtable
@@ -2327,7 +2327,7 @@
 	vt_board_unregister(board);
 } //}}}
 
-static int __devinit openpci_probe_board(struct pci_dev *pdev,
+static int openpci_probe_board(struct pci_dev *pdev,
 					 const struct pci_device_id *ent)
 { //{{{
 	struct openpci *card;
@@ -2562,7 +2562,7 @@
 	}
 } //}}}
 
-static void __devexit openpci_remove_board(struct pci_dev *pdev)
+static void openpci_remove_board(struct pci_dev *pdev)
 { //{{{
 	struct openpci *card = pci_get_drvdata(pdev);
 	unsigned long flags;
--- vpb-driver-4.2.56/src/vtcore/vtopenswitch.c~	2013-04-11 19:11:39.000000000 +0200
+++ vpb-driver-4.2.56/src/vtcore/vtopenswitch.c	2013-06-05 22:56:45.936416073 +0200
@@ -273,14 +273,14 @@
 };
 MODULE_DEVICE_TABLE(pci, openswitch_idtable);
 
-static int  __devinit openswitch_probe_board(struct pci_dev *pdev,
+static int  openswitch_probe_board(struct pci_dev *pdev,
 					     const struct pci_device_id *ent);
-static void __devexit openswitch_remove_board(struct pci_dev *pdev);
+static void openswitch_remove_board(struct pci_dev *pdev);
 
 static struct pci_driver vtopenswitch_driver = {
 	.name     = NAME,
 	.probe    = openswitch_probe_board,
-	.remove   = __devexit_p(openswitch_remove_board),
+	.remove   = openswitch_remove_board,
 	.suspend  = NULL,
 	.resume   = NULL,
 	.id_table = openswitch_idtable
@@ -1582,7 +1582,7 @@
 	return RET_OK;
 } //}}}
 
-static int __devinit openswitch_probe_board(struct pci_dev *pdev,
+static int openswitch_probe_board(struct pci_dev *pdev,
 					    const struct pci_device_id *ent)
 { //{{{
 	struct openswitch *card;
@@ -1858,7 +1858,7 @@
 	set_audio_off( card, port );
 } //}}}
 
-static void __devexit openswitch_remove_board(struct pci_dev *pdev)
+static void openswitch_remove_board(struct pci_dev *pdev)
 { //{{{
 	struct openswitch *card = pci_get_drvdata(pdev);
 	int    i = 0;
